_posts/2023-07-10-introducing-extensions-for-opensearch.md

```diff
@@ -0,0 +1,127 @@
+---
+
+_Extensions_ is a new experimental feature in OpenSearch 2.9 that allows you to extend OpenSearch and independently scale workloads without impacting cluster availability. In this blog post, we will introduce extensions and compare them to plugins. Using a 36-node cluster running a machine learning algorithm that performs high-cardinality anomaly detection, we'll demonstrate how we have achieved a cost reduction of 33% per data node, with performance matching that of a plugin and the only added cost of one extension node.
+
+## The old way: Plugins
+
+OpenSearch is a fork of Elasticsearch 7.10, which offered extending features through plugins. Until now, plugins were the best way to extend features in OpenSearch, and we’ve written a [blog post](https://opensearch.org/blog/plugins-intro/) to help you understand how they work. The OpenSearch project currently maintains 17 plugins, and we have learned a lot developing and operating them at scale over the last two years. In such, we’ve encountered many bottlenecks in the existing plugin architecture.
+
+Plugins are class-loaded into memory during OpenSearch bootstrap and therefore run within the OpenSearch process. This leads to the following three architectural problems:
+
+1. Plugins require rigid version compatibility with OpenSearch. Every patch version change in OpenSearch core requires the plugin to be recompiled and rereleased.
+2. Plugins are not isolated, so they can fatally impact the OpenSearch cluster and cannot scale independently. 
+3. Having to manage the dependencies of all the components together makes it challenging to include 20+ plugins in a single product distribution.
+
+## The new way: Extensions
+
+Unlike plugins, extensions run as a sidecar process or on a remote node, are isolated, and have cleaner, well-defined interfaces that interact with OpenSearch core. These interfaces are versioned (they follow semantic versioning) and are guaranteed to work across minor and patch versions, providing the future possibility of being compatible with multiple major versions of OpenSearch.
+
+The extensions architecture is illustrated in the following figure. For details, see [extension design documentation](https://opensearch-project.github.io/opensearch-sdk-java/DESIGN.html).
+
+![Extensions for OpenSearch](/assets/media/blog-images/2023-07-10-introducing-extensions-for-opensearch/extensions.png){: .img-fluid}
+
+## Experimental SDK launch
+
+The first step on our [extensibility roadmap](https://opensearch.org/blog/technical-roadmap-opensearch-extensibility/) is to launch a developer SDK that makes the experience of extending OpenSearch easier than with plugins. Today we are launching a Java version, [opensearch-sdk-java](https://github.com/opensearch-project/opensearch-sdk-java), that works with OpenSearch 2.9 and later. To turn on the experimental extensions feature, set `opensearch.experimental.feature.extensions.enabled` to `true` in your `opensearch.yml` file.
```

Comment:
We should include all the options to enable the experimental feature or just provide [this](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/DEVELOPER_GUIDE.md#enable-the-extensions-feature-flag) link.

Reply:
Done, thanks!